### PR TITLE
Code changes for TM to use latest profiles

### PR DESCRIPTION
**Pull request by sarangan12**
@vishrutshah @veronicagg @salameer Please review the modified traffic manager sample.

### CODE REVIEW

**Review by vishrutshah — DISMISSED** [review]
@sarangan12  Changes LGTM! 

Let's wait fro the release and update Gemfile to the released version or meanwhile if we want to push this out, let's update gemfile to point it directly to the `azure-stack` branch.

**Review by veronicagg — CHANGES_REQUESTED** [review]

**Review by devigned — COMMENTED** [review]
The module path seems odd. I would have expected `Azure::TrafficManager::Management::Profile_Latest` to be something like `Azure::Profiles::{profile_name}::TrafficManager::Mgmt`. Is the module path this way because it's in the resource provider gem?

**Review by devigned — DISMISSED** [review]
```ruby
# Profile API version for Traffic manager
TrafficManagerMgmt = Azure::TrafficManager::Profiles::v2016_09_01::Mgmt
tm_client = TrafficManagerMgmt::Client.new(options)
Models = TrafficManagerMgmt::Models

# Specific API version for TrafficManager
SpecificTM = Azure::TrafficManager::Mgmt::Api_2015_12_31
specific_tm_client = SpecificTM::TrafficManagerClient.new(options)
SpecificModels = SpecificTM::Models

# Usage from the top level azure_sdk rollup gem via a profile
TrafficManagerMgmt = Azure::Profiles::v2016_09_01::TrafficManager::Mgmt
tm_client = TrafficManagerMgmt::Client.new(options)
Models = TrafficManagerMgmt::Models

# Usage from the top level azure_sdk rollup gem via the Latest profile
TrafficManagerMgmt = Azure::Profiles::Latest::TrafficManager::Mgmt
tm_client = TrafficManagerMgmt::Client.new(options)
Models = TrafficManagerMgmt::Models
```

File: example.rb [hunk 1]
```diff
@@ -4,6 +4,11 @@
 require 'azure_mgmt_resources'
 require 'azure_mgmt_traffic_manager'
 
+TrafficManager =  Azure::TrafficManager::Management::Profile_Latest
```
[the remark this answers is not in the document]
Comment (sarangan12) [reply example.rb]: Changed it to a specific version

File: example.rb [hunk 2]
```diff
@@ -51,27 +59,27 @@ def run_example
...
   end
-  param.monitor_config = Azure::ARM::TrafficManager::Models::MonitorConfig.new.tap do |monitor_config|
+  param.monitor_config = TrafficManagerModels::MonitorConfig.new.tap do |monitor_config|
     monitor_config.protocol = 'HTTP'
     monitor_config.port = 80
     monitor_config.path = '/sample_monitor_page'
   end
 
-  profile = traffic_manager_client.profiles.create_or_update(RESOURCE_GROUP_NAME, PROFILE_NAME, param)
+  profile = traffic_manager_client.traffic_manager.profiles.create_or_update(RESOURCE_GROUP_NAME, PROFILE_NAME, param)
```
[the remark this answers is not in the document]
Comment (sarangan12) [reply example.rb]: No It is not under the client

File: example.rb [hunk 3]
```diff
@@ -28,18 +33,21 @@ def run_example
...
   # traffic manager client
-  traffic_manager_client = Azure::ARM::TrafficManager::TrafficManagerManagementClient.new(credentials)
-  traffic_manager_client.subscription_id = subscription_id
+  traffic_manager_client = TrafficManager::Client.new(options)
 
   #
   # Register subscription for 'Microsoft.KeyVault' namespace
   #
-  provider = resource_client.providers.register('Microsoft.Network')
+  provider = resource_client.resources.providers.register('Microsoft.Network')
```
[the remark this answers is not in the document]
Comment (sarangan12) [reply example.rb]: No It is not under the client

**Comment by veronicagg** [comment]
Shouldn't we be updating gem versions at https://github.com/Azure-Samples/traffic-manager-ruby-manage-profiles/blob/28983925d60fcce0a3bed374405552960469f434/Gemfile#L9?

**Comment by sarangan12** [comment]
@veronicagg Gemfile has been modified

**Comment by veronicagg** [comment]
@sarangan12 I think we should wait on merging all the updates to samples, until the gems are released, thanks!

**Review by veronicagg — APPROVED** [review]

**Pull request merged**